Add UBSan sample

voltur01 opened the pull request.

mplatings commented on samples/src/cpp-baremetal-semihosting-ubsan/libubsan.cpp:
```diff
@@ -0,0 +1,54 @@
+    HANDLER(nullability_return, "nullability-return")
+    HANDLER(pointer_overflow, "pointer-overflow")
+    HANDLER(cfi_check_fail, "cfi-check-fail")
+}
```
terminating newline

mplatings commented on samples/src/cpp-baremetal-semihosting-ubsan/libubsan.cpp:
```diff
@@ -0,0 +1,54 @@
+extern "C" {
+    #include <stdio.h>
```
Having includes in an `extern "C"` block seems like something to avoid.

mplatings reviewed (changes requested):
Can this be added to the list of samples in CMakeLists.txt, to ensure it doesn't bitrot?

voltur01 commented:
Thank you for the review: I fixed the year and added to the list of samples as in the other sample; also updated libubsan.cpp as per your suggestions.
